lib/generators.nix

```
@@ -14,29 +14,73 @@
  * Documentation in the manual, #sec-generators
  */
 { lib }:
-with (lib).trivial;
-let
-  libStr = lib.strings;
-  libAttr = lib.attrsets;
-
-  inherit (lib) isFunction;
-in
 
-rec {
+let
+  inherit (lib)
+    addErrorContext
+    assertMsg
+    attrNames
+    concatLists
+    concatMapStringsSep
+    concatStrings
+    concatStringsSep
+    const
+    elem
+    escape
+    filter
+    flatten
+    foldl
+    functionArgs  # Note: not the builtin; considers `__functor` in attrsets.
+    generators
+    gvariant
```

`generators` is not used :)

```suggestion
    gvariant
```

`gvariant` here is a bit out of place since it's a sublibrary with nested attributes, but I don't see a problem with it.

---

> gvariant here is a bit out of place since it's a sublibrary with nested attributes, but I don't see a problem with it.

It's just a name 🤷🏻 As previously mentioned, the rhyme is a little hard to find.